drivers/entropy/entropy_nrf5.c

```diff
@@ -272,44 +233,44 @@ static int entropy_nrf5_get_entropy_isr(struct device *dev, u8_t *buf, u16_t len
 	struct entropy_nrf5_dev_data *dev_data = DEV_DATA(dev);
 	u16_t cnt = len;
 
-	if (!(flags & ENTROPY_BUSYWAIT)) {
-		return get((struct rand *)dev_data->isr, len, buf);
+	if (likely((flags & ENTROPY_BUSYWAIT) == 0)) {
+		return rng_pool_get((struct rng_pool *)dev_data->isr, buf, len);
```

Comment:
In this case `len` is not checked against 0 but it is in all other cases (`ENTROPY_BUSYWAIT` and `entropy_nrf5_get_entropy`). It would be good to keep consistency.

Reply:
The `rng_pool_get()` works correctly if len = 0. Here, the check is skipped, as we have to be really fast.

Reply:
I propose another optimization then. Since we've got only one entropy device here, there is no need to use `DEV_DATA(dev)` in this function. You could use `entropy_nrf5_data` directly and save a few cycles more.

Reply:
Good point. Optimization added.